python/sglang/srt/managers/schedule_batch.py

```diff
@@ -1047,11 +1047,23 @@ def filter_batch(
         keep_indices: Optional[List[int]] = None,
     ):
         if keep_indices is None:
-            keep_indices = [
-                i
-                for i in range(len(self.reqs))
-                if not self.reqs[i].finished() and self.reqs[i] is not being_chunked_req
-            ]
+            keep_indices = []
+            spec_index, abort_count = 0, 0
+            spec_keep_indices = []
+            for i in range(len(self.reqs)):
+                finished_reason = self.reqs[i].finished_reason
+                if (
+                    not self.reqs[i].finished()
+                    and self.reqs[i] is not being_chunked_req
+                ):
+                    keep_indices.append(i)
+                if finished_reason is not None and isinstance(
+                    finished_reason, FINISH_ABORT
+                ):
+                    abort_count += 1
+                if not self.reqs[i].finished():
```

Comment:
Why can't the code under this if statement be placed under the if statement of line 1055?